e2e/cypress/integration/multi_team_and_dm/join_open_team_from_dm_spec.js

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// Stage: @prod
+// Group: @multi_team_and_dm
+
+describe('Join an open team from a direct message link', () => {
+    let openTeam;
+    let testUserInOpenTeam;
+
+    let secondTestTeam;
+    let testUserOutsideOpenTeam;
+
+    before(() => {
+        cy.apiCreateTeam('mmt452-second-team', 'mmt452-second-team', 'I', true).then(({team}) => {
+            secondTestTeam = team;
+
+            // # Create pool of users for tests
+            cy.apiCreateUser().then(({user}) => {
+                testUserOutsideOpenTeam = user;
+                cy.apiAddUserToTeam(secondTestTeam.id, testUserOutsideOpenTeam.id);
+            });
+        });
+
+        cy.apiCreateTeam('mmt452-open-team', 'mmt452-open-team', 'O', true).then(({team}) => {
+            openTeam = team;
+
+            // # Allow any user with an account on this server to join this team
+            cy.apiPatchTeam(openTeam.id, {
+                allow_open_invite: true,
+            });
+
+            // # Create pool of users for tests
+            cy.apiCreateUser().then(({user}) => {
+                testUserInOpenTeam = user;
+                cy.apiAddUserToTeam(openTeam.id, testUserInOpenTeam.id);
+
+                // # Login as test user
+                cy.apiLogin(testUserInOpenTeam);
+            });
+        });
+    });
+
+    it('MM-T452 User with no teams should be able to join an open team from a link in direct messages', () => {
+        // # View 'off-topic' channel
+        cy.visit(`/${openTeam.name}/channels/off-topic`);
+
+        // * Expect channel title to match title passed in argument
+        cy.get('#channelHeaderTitle').
+            should('be.visible').
+            and('contain.text', 'Off-Topic');
+
+        // # Copy full url to channel
+        cy.url().then((openTeamChannelUrl) => {
+            // # From the 'Direct Messages' menu, send the URL of the public channel to the user outside the team
+            sendDirectMessageToUser(testUserOutsideOpenTeam, openTeamChannelUrl);
+
+            // # Logout as test user
+            cy.apiLogout();
+
+            // # Login as user outside team
+            cy.apiLogin(testUserOutsideOpenTeam);
```

Comment:
I suggests to do `cy.reload()` after logging in as another user to take effect the state related to such current session.

Reply:
`cy.reload()` might be required in this case because it the test fails for me without it.